source/game/field/obj/ObjectKinoko.hh

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "egg/math/Vector.hh"
+#include "game/field/obj/ObjectCollidable.hh"
+#include "game/field/obj/ObjectKCL.hh"
+
+namespace Field {
+
+enum class KinokoType : u16 {
+    Light = 0,
+    Dark = 1,
+};
+
+/// @brief The class that's responsible for loading the collision data for the mushroom objects
+class ObjectKinokoStem : public ObjectCollidable {
+public:
+    ObjectKinokoStem(const System::MapdataGeoObj &params, KinokoType type);
+    ~ObjectKinokoStem() override;
+
+    [[nodiscard]] const char *getKclName() const override;
+
+private:
+    KinokoType m_type;
+};
+
+/// @brief The base class for a mushroom object with normal road properties, for the most part
```

Comment:
I don't think this is a base class. It's never inherited from, right?

Reply:
Both ``ObjectKinokoUd`` and ``ObjectKinokoBend`` inherit from it

Reply:
Right, i guess i was thinking that it *could* be used as a base class so that's why i worded it that way